test/dist_optimizer_test.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import unittest
+
+from dist_utils import INIT_METHOD_TEMPLATE, dist_init
+import torch
+import torch.distributed.autograd as dist_autograd
+import torch.distributed.optimizer.dist_optimizer as dist_optimizer
```

Comment:
The corresponding package and class is called `torch.optim.Optimizer`, do we want to keep the same convention here? i.e., `torch.distributed.optim.Optimizer`.